triton-vm/src/domain.rs

```diff
@@ -1,96 +1,100 @@
 use std::marker::PhantomData;
-use std::ops::MulAssign;
+use std::ops::{Mul, MulAssign};
 
 use twenty_first::shared_math::b_field_element::BFieldElement;
 use twenty_first::shared_math::polynomial::Polynomial;
 use twenty_first::shared_math::traits::{FiniteField, ModPowU32};
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct FriDomain<FF> {
+pub struct Domain<FF> {
     pub offset: BFieldElement,
-    pub omega: BFieldElement,
+    pub generator: BFieldElement,
     pub length: usize,
-    _field: PhantomData<FF>,
+    _finite_field: PhantomData<FF>,
 }
```

Comment:
Your realization is spot on. As of this PR, the STARK prover uses three different domains:
1. the trace domain (used to be “ο-domain”) (only represented as a `Domain` object inside `dual_low_degree_extension`)
2. the arithmetic domain (new)
3. the FRI domain (used to be “ω-domain”)

The verifier is unaffected by this change.

It is probably worth explaining this somewhere (more visible than here). Do you have a good idea on where that could be?

Reply:
Also, I concur with the need for a less generic name. The name `TraceDomain` is very misleading, because it only refers to one of the three possibilities. `AirDomain` might work well, thank you for the suggestion.

Reply:
The chosen name is `ArithmeticDomain`.